exasol/analytics/audit/audit.py

```diff
@@ -0,0 +1,74 @@
+from inspect import cleandoc
+from typing import List
+
+from exasol.analytics.audit.columns import AuditColumns
+from exasol.analytics.query_handler.query.select import (
+    AuditQuery,
+    ModifyQuery,
+    SelectQueryWithColumnDefinition,
+)
+from exasol.analytics.schema import (
+    Column,
+    SchemaName,
+    TableLikeName,
+    TableLikeNameImpl,
+    decimal_column,
+    timestamp_column,
+    varchar_column,
+)
+
+
+class TableDescription:
+    """
+    This class describes an SQL table by its attributes
+    * table (instance of (TableLikeName)) defining the name and optionally the
+      database schema of the SQL table
+    * columns the names and types of the columns of the SQL table
+
+    The class also offers a property `create` which renders the attributes
+    into an SQL CREATE TABLE statement. The class is tested together with
+    AuditTable in an integration test.
+    """
+
+    def __init__(self, table: TableLikeName, columns: List[Column]):
+        self.table = table
+        self.columns = {c.name.name: c for c in columns}
+
+    @property
+    def create(self):
+        columns = ",\n  ".join(c.for_create for c in self.columns.values())
+        return cleandoc(
+            f"""
+            CREATE TABLE IF NOT EXISTS {self.table.fully_qualified} (
+              {columns}
+            )
+            """
+        )
+
+
+class AuditTable(TableDescription):
+    def __init__(self, db_schema: str):
+        super().__init__(
+            table=TableLikeNameImpl("AUDIT_LOG", SchemaName(db_schema)),
+            columns=AuditColumns.all,
+        )
+
+
+def status_query(query: ModifyQuery) -> AuditQuery:
+    if query.modifies_row_count:
+        column = AuditColumns.ROWS_COUNT
+        table_name = query.db_object_ref.fully_qualified
+        column_name = column.name.fully_qualified
+        count_query = f"SELECT COUNT(1) AS {column_name} FROM {table_name}"
+        output_columns = [column]
+    else:
+        count_query = "SELECT 1"
+        output_columns = []
```

Comment:
this is not legal, you define a query with one output column, but don't specify one. I also don't understand for what you need the second case.

Reply:
AFAIK, A modify query potentially can also indicate to modify a SCHEMA, VIEW, FUNCTION, or SCRIPT.
In these cases, we cannot count the number of rows at all. 

AFAIK, A modify query potentially can also have DbOperationType ALTER, DROP, UPDATE, EXECUTE.
In these cases we the number of rows is expected to remain unchanged and we don't need to count them.

Reply:
I will move this change to a separate PR.

Reply:
see next push